lib/services/awsAdapter.js

```diff
@@ -1,13 +1,47 @@
 "use strict";
 
-var Bluebird = require("bluebird");
-
-function AWSAdapter (ec2) {
+var Bluebird        = require("bluebird");
+var SshAdapter      = require("./sshAdapter");
+var Instance        = require("../models/Instance");
+var InstanceAdapter = require("./instanceAdapter");
+var Joi             = require("joi");
+var AWS             = require("aws-sdk");
 
+function AWSAdapter (options) {
 	var tags;
 	var instanceID;
+	var sshAdapter;
+	var instances;
+	var serverLog;
+	var ec2;
+	var schema = Joi.object().keys({
+		sshAdapter : Joi.object().type(SshAdapter).required(),
+		ec2        : Joi.object().type(AWS.EC2).required(),
+		serverLog  : Joi.any().required(),
+		instances  : Joi.object().type(InstanceAdapter).required()
+	});
+
+	if (options) {
+		if (options.sshAdapter) {
+			sshAdapter = options.sshAdapter;
+		}
+		else if (options.ec2) {
+			sshAdapter = new SshAdapter(options.ec2);
+		}
+		instances  = options.instances;
+		serverLog  = options.serverLog;
+		ec2        = options.ec2;
+	}
+
+	Joi.validate(options, schema, function (error) {
```

Comment:
It probably makes more sense to use the synchronous form of `Joi.validate()` here since this is a synchronous function. Check out the code examples in the [`validate()` documentation](https://github.com/hapijs/joi#validatevalue-schema-options-callback).